config/examples/wrapped_json.toml

```
@@ -14,25 +15,19 @@ data_dir = "/var/lib/vector"
   ignore_older = 86400 # 1 day
 
 # Parse the data as JSON
-# Docs: https://vector.dev/docs/reference/transforms/json_parser
-[transforms.parse_root_json]
+# Docs: https://vector.dev/docs/reference/transforms/remap
+[transforms.parse_json]
   inputs       = ["logs"]
-  type         = "json_parser"
-  field        = "message" # default
-
-[transforms.parse_parent_json]
-  inputs       = ["parse_root_json"]
-  type         = "json_parser"
-  field        = "parent"
-
-[transforms.parse_child_json]
-  inputs       = ["parse_parent_json"]
-  type         = "json_parser"
-  field        = "child"
+  type         = "remap"
+  source       = '''
+  . = parse_json!(string!(del(.message)))
+  . = parse_json!(string!(del(.parent)))
+  . = parse_json!(string!(del(.child)))
```

```suggestion
  . |= parse_json!(string!(del(.message)))
  . |= parse_json!(string!(del(.parent)))
  . |= parse_json!(string!(del(.child)))
```

This will also leave the original `.message`, `.parent`, and `.child` paths on the event.

---

Hmmm.... but if we're simply merging (`|`), how is that any different than direct assignment?

---

Direct assignment would remove the `.parent`, and `.child` fields. As to the merge assignment behaviour - https://github.com/vectordotdev/vector/issues/13205

---

🤔 there's probably a `map_value` iteration we could do there as well - `for each value parse as json and merge into root`

---

Addressed in b2afb1ce0.